mongodump/metadata_dump.go

```diff
@@ -44,14 +42,7 @@ func (dump *MongoDump) dumpMetadata(intent *intents.Intent, buffer resettableOut
 	}
 
 	// The collection options were already gathered while building the list of intents.
-	// We convert them to JSON so that they can be written to the metadata json file as text.
-	if intent.Options != nil {
-		if meta.Options, err = bsonutil.ConvertBSONValueToLegacyExtJSON(intent.Options); err != nil {
-			return fmt.Errorf("error converting collection options to JSON: %v", err)
-		}
-	} else {
-		meta.Options = nil
-	}
+	meta.Options = intent.Options
```

Comment:
I guess a downside of this approach might be that the error reporting is less specific if we run into a problem? 
before we could pinpoint that the issue is in the collection options, now we maybe can't? (not sure what the marshaling error would look like.) but maybe that's not very important anyway.

if we're going to do it this way, should we consider more strongly typing the values we store in the `Metadata`? for example, rather than just saying `meta.Options` is an interface it could be its actual type (looks like a `bson.D`?)

Reply:
Done in new commit